### PR TITLE
Neon adjustments

### PR DESCRIPTION
**Pull request by lgalabru**

### CODE REVIEW

File: src/net/db.rs [hunk 1]
```diff
@@ -112,9 +112,17 @@ impl fmt::Debug for LocalPeer {
 }
 
 impl LocalPeer {
-    pub fn new(network_id: u32, parent_network_id: u32, port: u16, key_expire: u64, data_url: UrlString) -> LocalPeer {
+    pub fn new(network_id: u32, parent_network_id: u32, port: u16, key_expire: u64, data_url: UrlString, seed: Vec<u8>) -> LocalPeer {
+
+        let mut re_hashed_seed = seed;
```
Comment (jcnelson) [inline src/net/db.rs:117]: Why not just provide the private key?

Comment (lgalabru) [reply src/net/db.rs:117]: It was insanely straightforward :).
You prefer a private key? hex? wif?

**Comment by jcnelson** [comment]
I'm going to add a couple things to this PR

File: src/net/db.rs (hunk 1 above)
Comment (jcnelson) [reply src/net/db.rs:117]: Don't worry about it; I've already changed it in my local copy.  It'll just take the private key, and the testnet node will generate it from its seed when instantiating the peer DB.

**Comment by jcnelson** [comment]
My changes do the following:

* Generate the node private key in the testnet, and pass the private key to the PeerDB.  Leaves seed usage and key generation to the testnet.
* Adds `p2p_address` configuration directive, which takes the public IP and port to which p2p message should be sent.  This is what will be advertized to nodes via `HandshakeAccept` messages, so it must be a routable address.
* Changes `TESTNET_CHAIN_ID` to `0x80000000`.  In general, the only chain ID that should be 0 ought to be Bitcoin; every Stacks chain should have a non-zero chain ID.

File: src/net/db.rs (hunk 1 above)
Comment (lgalabru) [reply src/net/db.rs]: gotcha, I thought you wanted to pass the private key in the config, instead of a seed.

File: testnet/src/neon_node.rs [hunk 2]
```diff
@@ -28,15 +28,18 @@ use stacks::util::vrf::VRFPublicKey;
...
+use stacks::util::secp256k1::Secp256k1PrivateKey;
 use stacks::net::NetworkResult;
+use stacks::net::PeerAddress;
 use std::sync::mpsc::{sync_channel, TrySendError, SyncSender, Receiver};
 use crate::burnchains::bitcoin_regtest_controller::BitcoinRegtestController;
 use crate::ChainTip;
 
 use stacks::burnchains::BurnchainSigner;
 
 pub const TESTNET_CHAIN_ID: u32 = 0x00000000;
```
Comment (lgalabru) [inline testnet/src/neon_node.rs:41]: This `TESTNET_CHAIN_ID` probably need to change?

Comment (jcnelson) [reply testnet/src/neon_node.rs:41]: Yes, it should.  To `0x80000000`

Comment (jcnelson) [reply testnet/src/neon_node.rs]: Fixed

**Comment by lgalabru** [comment]
Looks good to me, thank you @jcnelson!

**Pull request merged**